src/sequential_cursor.rs

```diff
@@ -50,15 +52,17 @@ impl<'s> std::io::Read for SeqCursor<'s> {
             acc_offset += chunk_len;
             if self.position < acc_offset as u64 {
                 let remaining_in_chunk = (acc_offset as u64 - self.position) as usize;
+                let start = (chunk_len - remaining_in_chunk) as u64;
                 let fin = std::cmp::min(total_read + remaining_in_chunk, buf.len());
+                cursor.seek(SeekFrom::Start(start))?;
```

Comment:
We could seek to `0` at the beginning in the loop

Reply:
Could you elaborate why this was closed as resolved?

Reply:
thanks @drahnr , let me explain here:
Your suggestion will fail when reading Seqcursor with a short buffer, cause it will always start from zero, I added another testcase for this.